chore: update deps and move vue to devDeps, fix #350

IlCallo opened the pull request:
Closes #350
Apparently the problem was due to the package installing `vue` as dependency, thus causing some kind of conflict
Moving it to devDeps and re-building the package with Vue 3.2 solved the problem for me, run tests and it's all green

neighborhood999 commented:
Hi @IlCallo 

Thank you for raising an issue and fix. Can you try to remove `yarn.lock` and re-install dependencies again to confirm this problem?

IlCallo commented:
I did it multiple times when testing the problem, to no avail

IlCallo commented:
@neighborhood999 the problem is still present with latest releases
I rebased my branch on latest changes and added vue as peerDependency too

The problem may also be caused by this
![image](https://user-images.githubusercontent.com/10036108/156197457-32414a19-b959-4957-96aa-1856e5b0a1f0.png)
![image](https://user-images.githubusercontent.com/10036108/156197624-4e732cec-1c8c-4d98-970e-6f1f24f95cb6.png)

2 version of Vue compiler are present, one required by Vue dep and the one is direct
Vue3 actually serves it's compiler via `vue/compiler-sfc` since 3.2, but apparently the used rollup plugin search for the direct dependency
Vue libraries compiled with compiler version <3.2 aren't compatible with apps using Vue 3.2, the duplicated compiler version in the yarn.lock may be the cause of my issue

Having vue as direct dependency is against best practices and can cause problems, can you test this and merge it if everything still works for you?
Otherwise I'll just republish it with the fix under my company scope

neighborhood999 commented on package.json:
```diff
@@ -82,7 +81,7 @@
     "@vue/cli-plugin-babel": "4.5.15",
     "@vue/cli-service": "4.5.15",
-    "@vue/compiler-sfc": "3.1.5",
+    "@vue/compiler-sfc": "^3.2.31",
```
Can you specify the version directly? Same as the below.

```suggestion
    "@vue/compiler-sfc": "3.2.31",
```

neighborhood999 commented:
> @neighborhood999 the problem is still present with latest releases I rebased my branch on latest changes and added vue as peerDependency too
> 
> The problem may also be caused by this ![image](https://user-images.githubusercontent.com/10036108/156197457-32414a19-b959-4957-96aa-1856e5b0a1f0.png) ![image](https://user-images.githubusercontent.com/10036108/156197624-4e732cec-1c8c-4d98-970e-6f1f24f95cb6.png)
> 
> 2 version of Vue compiler are present, one required by Vue dep and the one is direct Vue3 actually serves it's compiler via `vue/compiler-sfc` since 3.2, but apparently the used rollup plugin search for the direct dependency Vue libraries compiled with compiler version <3.2 aren't compatible with apps using Vue 3.2, the duplicated compiler version in the yarn.lock may be the cause of my issue
> 
> Having vue as direct dependency is against best practices and can cause problems, can you test this and merge it if everything still works for you? Otherwise I'll just republish it with the fix under my company scope

Sorry for the reply lately. I thought moving the Vue as the `peerDependencies` is make sense. I left a review, please help change and I will publish the beta version for test.

IlCallo commented:
Done

neighborhood999 reviewed (approved):
Thanks, will publish the beta version for v3

neighborhood999 commented:
Published `vue-signature-pad@3.0.2-beta.3`, please give it try.

IlCallo commented:
Yep, it's working
Pay attention, as you're releasing beta packages under the "latest" tag, instead of "beta" one

neighborhood999 commented:
Thank you for testing. You're right, it's my fault, I forgot to add the `beta` tag when I publish.
